src/main/scala/org/apache/sysml/api/ml/LogisticRegression.scala

```diff
@@ -160,8 +167,8 @@ class LogisticRegressionModel(
     val rawPred = outNew.getDF(df.sqlContext, "rawPred", true).withColumnRenamed("C1", "rawPrediction").withColumnRenamed("ID", "ID2")
     var predictionsNProb = prob.join(pred, prob.col("ID").equalTo(pred.col("ID1"))).select("ID", "probability", "prediction")
     predictionsNProb = predictionsNProb.join(rawPred, predictionsNProb.col("ID").equalTo(rawPred.col("ID2"))).select("ID", "probability", "prediction", "rawPrediction")
-    val dataset1 = RDDConverterUtils.addIDToDataFrame(df, df.sqlContext, "ID")
-    dataset1.join(predictionsNProb, dataset1.col("ID").equalTo(predictionsNProb.col("ID")))
+    val dataset1 = RDDConverterUtilsExt.addIDToDataFrame(df,"dfindex",0)
```

Comment:
@dusenberrymw Yes, since the ID correspond to row indexes (i.e. X[i,]) they are one-based.

@Wenpei Can you explain where the zero-based IDs are coming in ? If one of our converter utils is returning a zero-based IDs, then its a bug and we should fix that instead of making start index settable.